emcc.py

```diff
@@ -3168,15 +3169,19 @@ def modularize():
         script_url = "import.meta.url"
       else:
         script_url = "typeof document !== 'undefined' && document.currentScript ? document.currentScript.src : undefined"
+        if shared.Settings.target_environment_may_be('node'):
+          script_url_node = "if (typeof __filename !== 'undefined') _scriptDir = _scriptDir || __filename;"
```

Comment:
Perhaps this should include additional checking for node? In shell.js we use
```javascript
typeof process === 'object' && typeof process.versions === 'object' && typeof process.versions.node === 'string'
```
(lengthier, but unambiguous). On the other hand, maybe `__filename` is unlikely to exist by coincidence?

Reply:
I figure doing `_scriptDir = _scriptDir || __filename` is pretty safe. If `_scriptDir` is already valid it gets used, but if it's not `__filename` is used.
In the case where `_scriptDir` is not valid and `__filename` exists by coincidence then initialising the web workers is going to fail anyway.

Reply:
Makes sense, yeah. Looks ready to land then, aside from a test. Thanks!